test/lang.test.js

```diff
@@ -0,0 +1,71 @@
+const { describe, it } = require('mocha');
+const expect = require('chai').expect;
+const en = require('../lang/en.json');
+const { __ } = require('../lang/locales.js');
+const { SUPPORTED_LANGS } = require('../utils/constants.js');
+const LANGUAGE_CODES = Object.keys(SUPPORTED_LANGS);
+
+const size = Object.keys(en).length;
+
+describe('Localisation', function() {
+
+	for (const lang of LANGUAGE_CODES) {
+		// Skips English as it's the default language.
+		if (lang === 'en') continue;
+
+		// Gets the language file.
+		let locale;
+		try {
+			locale = require(`../lang/${lang}.json`);
+		}
+		catch (error) {
+			console.error(`Error | ${lang}.json doesn't exist!`);
+			locale = {};
+		}
+
+		describe(`# Language: ${lang.toUpperCase()}`, function() {
+
+			it(`Should have a language file "${lang}.json"`, function() {
+				expect(locale, `lang/${lang}.json`).to.be.an('object').and.not.empty;
+			});
+
+			it(`Should have ${size} translation keys`, function() {
+				expect(Object.keys(locale)).to.have.lengthOf(size);
+			});
+
+			it('Should have all en.json keys', function() {
+				const out = [];
+				for (const key in en) if (!(key in locale)) out.push(key);
+				expect(out.join(), `Missing keys from ${lang}.json`).to.equal([].join());
```

Comment:
```suggestion
				expect([].join(), `Missing keys from ${lang}.json`).to.equal(out.join());
```

Reply:
Why this change ? I don't understand.

Reply:
In my test the expected vs. actual output are in the wrong order. Expected would be empty and actual would be the expected key.